autogluon/utils/tabular/ml/learner/abstract_learner.py

```diff
@@ -452,16 +452,14 @@ def leaderboard(self, X=None, y=None, silent=False):
                 print(leaderboard)
         return leaderboard
 
-    # TODO: enable_fit_continuation must be set to True to be able to pass X and y as None in this function, otherwise it will error.
+    # TODO: cache_data must be set to True to be able to pass X and y as None in this function, otherwise it will error.
     # Warning: This can take a very, very long time to compute if the data is large and the model is complex.
-    # TODO: Warning: This is not memory safe, large datasets may result in OOM errors during feature importance calculation.
-    # TODO: Rewrite description before adding to Predictor
     # A value of 0.01 means that the objective metric error would be expected to increase by 0.01 if the feature were removed.
     # Negative values mean the feature is likely harmful.
     # model: model (str) to get feature importances for, if None will choose best model.
     # features: list of feature names that feature importances are calculated for and returned, specify None to get all feature importances.
     # raw: Whether to compute feature importance on raw original features or on the features used by the particular model.
-    def get_feature_importance(self, model=None, X=None, y=None, features: list = None, raw=True) -> Series:
+    def get_feature_importance(self, model=None, X=None, y=None, features: list = None, raw=True, silent=False) -> Series:
         if X is not None:
```

Comment:
should we print informative error message if the data are None and `cache_data` was not previously specified?

Reply:
Updated